### PR TITLE
List available paths on root path

### PR DESCRIPTION
**Pull request by matej-g**
This PR adds a new handler for `/` to return a list of available API paths. The purpose of this is to improve discoverability of the API paths and to improve the user experience.

The changes also including bumping the `signal` packaged user for internal server and adds some missing docs.

### CODE REVIEW

File: authentication/authentication.go [hunk 1]
```diff
@@ -102,6 +102,7 @@ func (ah *ProviderManager) InitializeProvider(config map[string]interface{},
 	return authCh
 }
 
+// Middleware returns an authentication middleware for a tenant.
```
Comment (tareqmamari) [inline authentication/authentication.go:105]: Thanks for fixing these @matej-g. Middleware should be Middlewares

Comment (matej-g) [reply authentication/authentication.go:105]: Ah missed that, thanks! Will fix it.

**Review by ianbillett — COMMENTED** [review]
@matej-g please can you provide an example of the response that we get from this change?

To @squat 's point in the previous PR - we don't want to be leaking any sensitive information to the world.

**Review by moadz — APPROVED** [review]
LGTM

**Comment by matej-g** [comment]
> @matej-g please can you provide an example of the response that we get from this change?
> 
> To @squat 's point in the previous PR - we don't want to be leaking any sensitive information to the world.

It basically lists the routes but only as patterns, i.e. the actual tenant names etc. are not substituted. Example:
```
{
  "paths": [
    "/api/logs/v1/{tenant}/*",
    "/api/metrics/v1/{tenant}/*",
    "/api/v1/{tenant}/*",
    "/{tenant}"
  ]
}
```

**Review by ianbillett — APPROVED** [review]
LGTM - nice stuff

**Review by periklis — COMMENTED** [review]
Looks good to me. Is it possible to list the valid HTTP verbs per path?

**Comment by matej-g** [comment]
> Looks good to me. Is it possible to list the valid HTTP verbs per path?

Hah, funnily enough when I tried to list it with methods, I'm seeing we simply accept _any_ method on any endpoint. I think we should be returning `405` for the ones which do not apply. I'm creating a ticket for this. I would then suggest to revisit this once we have the incorrect methods behaving correctly.

File: authentication/authentication.go (hunk 1 above)
Comment (squat) [reply authentication/authentication.go:105]: ping here

**Pull request merged**